image-cropper/src/app/file_selector/file_selector.rs

```diff
@@ -0,0 +1,156 @@
+                            let url = url.clone();
+                            let name = name.clone();
+                            Cropper::new(image, url, name)
+                        },
+                    )))
+                }
+            }
+        }
+    }
+
+    pub fn render(&self, props: &FileSelectorProps) -> RenderingTree {
+        const MARGIN: f32 = 16.0;
+        const BUTTON_HEIGHT: f32 = 36.0;
+        let button_wh = Wh {
+            width: props.screen_wh.width - (2.0 * MARGIN),
+            height: BUTTON_HEIGHT,
+        };
+        render([
+            render_background(&props.screen_wh),
+            translate(
+                MARGIN,
+                (props.screen_wh.height - BUTTON_HEIGHT) / 2.0,
+                render_file_select_dialog_open_button(button_wh),
+            ),
+        ])
+    }
+}
+
+fn render_background(wh: &Wh<f32>) -> RenderingTree {
+    namui::rect(RectParam {
+        x: 0.0,
+        y: 0.0,
+        width: wh.width,
+        height: wh.height,
+        style: RectStyle {
+            stroke: None,
+            fill: Some(RectFill {
+                color: Color::from_u8(36, 37, 42, 255),
+            }),
+            round: None,
+        },
+    })
+}
+
+fn create_image_input_element() -> HtmlInputElement {
+    let document = window()
+        .expect("failed to get window")
+        .document()
+        .expect("failed to get window.document");
+    let element = document
+        .create_element("input")
+        .expect("failed to create HTMLInputElement");
+    let input_element = wasm_bindgen::JsCast::dyn_into::<HtmlInputElement>(element)
+        .expect("failed to cast HTMLInputElement");
+    input_element.set_type("file");
+    input_element.set_accept("image/*");
+    input_element
+}
+
+fn set_file_handler(element: &HtmlInputElement, handler: fn(Vec<u8>, String, String)) {
+    element
+        .add_event_listener_with_callback(
+            "change",
+            Closure::wrap(Box::new(move |event: InputEvent| {
+                let target = event
+                    .target()
+                    .unwrap()
+                    .dyn_into::<HtmlInputElement>()
+                    .unwrap();
+                if let Some(files) = target.files() {
+                    if let Ok(files) = files.dyn_into::<FileList>() {
+                        if let Some(file) = files.item(0) {
+                            spawn_local(async move {
+                                let uint8array = Uint8Array::new(
+                                    &JsFuture::from(file.array_buffer())
+                                        .await
+                                        .expect("failed to read File as ArrayBuffer"),
+                                );
+                                let name = file.name();
+                                let url = Url::create_object_url_with_blob(file.as_ref())
+                                    .expect("Failed to create object url of image");
```

Comment:
This object url will be used at image saving stage to draw original image to temporary canvas.
And temporary canvas will be saved as image file.